lib/new_relic/transaction/plug.ex

```diff
@@ -80,4 +82,30 @@ defmodule NewRelic.Transaction.Plug do
       _ -> nil
     end
   end
+
+  defp maybe_report_queueing(conn) do
+    [
+      queue_start_us: extract_queue_start_us(conn)
+    ]
+    |> NewRelic.add_attributes()
+  end
+
+  @queue_duration_headers ["x-request-start", "x-queue-start", "x-middleware-start"]
+
+  defp extract_queue_start_us(%{req_headers: headers}),
+    do:
+      Enum.reduce_while(headers, nil, fn
+        {header, timestamp}, acc when header in @queue_duration_headers ->
+          case RequestQueueTime.timestamp_to_us(timestamp) do
+            {:ok, us} ->
+              {:halt, us}
+
+            {:error, reason} ->
+              Logger.debug(reason)
```

Comment:
There's a NewRelic logger that'd be preferable, though i'm not sure it's something i'd really want to log out

Reply:
I'm happy to remove it. I put it here to mirror the Ruby agent log. I could see someone wondering "why isn't my queueing metric reporting?"